demo-simple/src/main/java/com/novoda/downloadmanager/demo/VersionOneDatabaseCloner.java

```diff
@@ -0,0 +1,137 @@
+package com.novoda.downloadmanager.demo;
+
+import android.content.res.AssetManager;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.os.Handler;
+
+import com.novoda.downloadmanager.SqlDatabaseWrapper;
+import com.novoda.notils.exception.DeveloperError;
+import com.novoda.notils.logger.simple.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+class VersionOneDatabaseCloner {
+
+    private static final byte[] BUFFER = new byte[1024];
+
+    private static final String DATABASE_NAME = "downloads.db";
+    private static final String DOWNLOAD_FILE_NAME = "file.zip";
+    private static final String ORIGINAL_FILE_LOCATION_QUERY = "SELECT _data FROM Downloads WHERE _data IS NOT NULL";
+    private static final int ORIGINAL_FILE_LOCATION_COLUMN_INDEX = 0;
+
+    private final AssetManager assetManager;
+    private final Executor executor;
+    private final UpdateListener updateListener;
+    private final Handler updateHandler;
```

Comment:
This is basically the main thread right? Slightly weird having threads represented using both `Handler` and `Executor` but i'm not hugely bothered given the temporary nature of the code here 🤷‍♂️